### PR TITLE
LEAF 3457 - site designer homepage (review/comment only)

### PR DESCRIPTION
**Pull request by aerinkayne**
Multiple designs can be stored, and they can be deleted or copied.  Publication prompts for confirmation.

Adds Header design section with prosemirror WYSIWYG / markdown text options.

Adds nav with menu options, and various UI improvements.

Adds dedicated class, controller and table for design information.
Logs content change, draft/publish, create and delete events.

other
App and dialog organization improvements.
Removes (jQuery) ajax calls and replaces with fetch API calls.
User facing page is one page instead of homepage + search combo.

### CODE REVIEW

**Review by maxrichardjr — APPROVED** [review]

File: libs/js/vue-dest/site_designer/LEAF_Designer.js.LICENSE.txt
Comment (raphaelfretter) [inline libs/js/vue-dest/site_designer/LEAF_Designer.js.LICENSE.txt:1]: what is this license for?

**Review by raphaelfretter — CHANGES_REQUESTED** [review]

File: libs/js/vue-dest/site_designer/LEAF_Designer.js.LICENSE.txt
Comment (shaneodd) [reply libs/js/vue-dest/site_designer/LEAF_Designer.js.LICENSE.txt:1]: This package is getting pulled in, I am guessing  probably a wysiwyg editor of some sort probably pulled it in. npm is a wild ride sometimes. https://github.com/facebook/regenerator

**Review by shaneodd — APPROVED** [review]

File: libs/js/vue-dest/site_designer/LEAF_Designer.js.LICENSE.txt
Comment (aerinkayne) [reply libs/js/vue-dest/site_designer/LEAF_Designer.js.LICENSE.txt:1]: I am fairly certain all of these are due to using async await in the code and how Babel compiles it.  I was trying to proactively avoid ajax calls and used the fetch API instead.  Using async await there also simplified a couple things but (I realized later) also resulted in the regenerator runtime licenses.  If there were only one, I wouldn't care but I think I was getting 1 per view plus some mystery chunk files.  When I pick this up again, I am just going to nest the JS and hopefully not get too many callback doom pyramids.

**Review by jampaul3 — APPROVED** [review]

**Comment by aerinkayne** [comment]
Adding a comment for visibility / future sanity

Current master should be used for any src/common conflicts.  Site Designer dialogs should be updated to use dialogData the way it is used in the Form Editor App.
Use of async-await resulted in the 'generator-runtime' LICENSE files - replacing it with chaining should prevent these.

**Pull request closed without merging**